### PR TITLE
IPC service support

### PR DESCRIPTION
**Pull request by kapi-no**
Migrated from the RPMsg service to the IPC service module which is used as inter-core transport mechanism for Bluetooth and 802.15.4 protocols on nRF53 devices.

### CODE REVIEW

**Review by hubertmis — APPROVED** [review]

**Comment by kapi-no** [comment]
This PR change is now part of the Upmerge PR: https://github.com/nrfconnect/sdk-nrf/pull/7050

This PR is now only for validation/verification purposes and will eventually be closed.

File: samples/nrf5340/multiprotocol_rpmsg/src/main.c [hunk 1]
```diff
@@ -224,6 +236,19 @@ void main(void)
 			NULL, NULL, NULL, K_PRIO_COOP(7), 0, K_NO_WAIT);
 	k_thread_name_set(&tx_thread_data, "HCI rpmsg TX");
 
+	/* Initialize IPC service instance and register endpoint. */
+	err = ipc_service_open_instance(hci_ipc_instance);
+	if (err) {
+		LOG_ERR("IPC service instance initialization failed: %d\n", err);
```
Comment (KAGA164) [inline samples/nrf5340/multiprotocol_rpmsg/src/main.c:242]:
```suggestion
		LOG_ERR("IPC service instance initialization failed: %d", err);
```

File: subsys/nrf_rpc/nrf_rpc_rpmsg.c [hunk 2]
```diff
@@ -31,27 +26,51 @@
...
+/* Semaphore used to synchronize endpoint binding.*/
+K_SEM_DEFINE(ipc_bound_sem, 0, 1);
 
 /* Upper level callbacks */
 static nrf_rpc_tr_receive_handler_t receive_callback;
 
-static int endpoint_id;
-static bool is_handshake_done;
+/* IPC service endpoint instance */
+struct ipc_ept nrf_rpc_ept;
```
Comment (KAGA164) [inline subsys/nrf_rpc/nrf_rpc_rpmsg.c:36]: static ?

Comment (KAGA164) [inline subsys/nrf_rpc/nrf_rpc_rpmsg.c:36]: static ?

File: subsys/nrf_rpc/nrf_rpc_rpmsg.c [hunk 3]
```diff
@@ -63,91 +82,37 @@ static int translate_error(int rpmsg_err)
...
-			}
-			rpmsg_service_send(endpoint_id, (uint8_t *)"", 0);
-			is_handshake_done = true;
-			NRF_RPC_INF("Handshake done");
-			event_handler(NRF_RPC_EVENT_CONNECTED, NULL, 0);
-		}
-		return RPMSG_SUCCESS;
+	err = ipc_service_open_instance(nrf_rpc_ipc_instance);
+	if (err < 0) {
+		NRF_RPC_ERR("IPC service instance initialization failed: %d\n", err);
```
Comment (KAGA164) [inline subsys/nrf_rpc/nrf_rpc_rpmsg.c:105]:
```suggestion
		NRF_RPC_ERR("IPC service instance initialization failed: %d", err);
```

**Review by KAGA164 — CHANGES_REQUESTED** [review]

**Comment by DatGizmo** [comment]
With the new Integration Test Specification process we don't need the github labels.
I will remove themI.

Right now this PR runs all Integration tests because it changes sdk-zephyr in west.yml and there is no test-spec.yml in that repo yet.
You can add changes to this PR [nrfconnect/sdk-zephyr#767](https://github.com/nrfconnect/sdk-zephyr/pull/767) to make the test spec here more targeted.

To get the test spec feature working correctly on this PR, you need to rebase it to the latest main to get the [.github/test-spec.yml](https://github.com/nrfconnect/sdk-nrf/blob/main/.github/test-spec.yml) file

Then run CI here to update the test spec.

[Information about the test spec feature](https://projecttools.nordicsemi.no/confluence/display/NCS/Test+Spec)
[CI-Help for issues](https://ci-help.nordicsemi.no/)

**Pull request closed without merging**